### PR TITLE
Patch CVE-2024-24786 in flannel

### PR DESCRIPTION
**Pull request by suresh-thelkar**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Patch CVE-2024-24786 in flannel and kata-containers-cc

###### Change Log  

- CVE-2024-24786

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-24786

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/flannel/flannel.spec [hunk 1]
```diff
@@ -21,12 +22,9 @@ BuildRequires:  kernel-headers
 Flannel is a simple and easy way to configure a layer 3 network fabric designed for Kubernetes.
 
 %prep
-%autosetup -p1
+%autosetup -p1 -a 1
 
 %build
-# create vendor folder from the vendor tarball and set vendor mode
```
Comment (jslobodzian) [inline SPECS/flannel/flannel.spec:27]: What's the reason for removing this?

Comment (jslobodzian) [reply SPECS/flannel/flannel.spec:27]: Actually I see that the -a 1 was added to handle this tar thing.  But is the --no-same-owner preserved by the -a 1 and/or was it even necessary to begin with.

File: SPECS/flannel/flannel.spec [hunk 2]
```diff
@@ -21,12 +22,9 @@ BuildRequires:  kernel-headers
 Flannel is a simple and easy way to configure a layer 3 network fabric designed for Kubernetes.
 
 %prep
-%autosetup -p1
+%autosetup -p1 -a 1
```
[the remark this answers is not in the document]
Comment (suresh-thelkar) [reply SPECS/flannel/flannel.spec:25]: This option -a 1 extracts source1 as well.

File: SPECS/flannel/flannel.spec (hunk 1 above)
Comment (suresh-thelkar) [reply SPECS/flannel/flannel.spec:27]: Keeping the following line of code in the %build section is overriding the vendor folder. This meant that any patches applied in the %prep section were lost. While this wasn't an issue previously due to the absence of patches for this package, we are now introducing a patch for the CVE. 

```
# create vendor folder from the vendor tarball and set vendor mode
tar -xf %{SOURCE1} --no-same-owner
```

To address this, I have removed the above line of code from the %build section and incorporated it into the %prep section. This change ensures that all setup-related activities are concentrated in the %prep section. 

Additionally, I have utilized the "-a 1" option for %autosetup, which extracts SOURCE1 as well. The command “%autosetup -p1 -a 1” is instrumental in achieving this setup consolidation. Under the hood, the extraction is done using rpmuncompress, which I believe has the same effect as using tar with the -no-same-owner option.

image

**Comment by suresh-thelkar** [comment]
@jslobodzian, I have addressed your comments.  Could you please go through it and let me know if there are any further review comments.

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**